autoscheduler/frontend/src/redux/reducers/schedules.ts

```diff
@@ -39,8 +40,12 @@ export interface RenameScheduleAction {
   index: number;
   name: string;
 }
+export interface SetScheduleAction {
+  type: 'SET_SCHEDULE';
+  schedules: Schedule[];
+}
```

Comment:
Current name makes it seem like it replaces a single schedule rather than all of them (also change the SET_SCHEDULES declaration obviously)
```suggestion
export interface SetSchedulesAction {
  type: 'SET_SCHEDULES';
  schedules: Schedule[];
}
```